### ISSUE TITLE
Which Version of contiki should I use？

### ISSUE THREAD

**Issue by LittleFatHero** [contiki-os/contiki#2554]
Hello, 
Recently，I found that a lot  of people choose to use contiki 2.7 instead of contiki 3.0, I was curious about this. For general experience，the newer the better, why they don't use  contiki 3.0, is it because of contiki 3.0 require more RAM of MCU or the code of contiki 3.0 is bigger than contiki 2.7 ?
truely hope someone could tell me about that, Thank You!

**Comment by ekawahyu** [comment]
There are a ton of changes actually :), but mainly, as you already have said, the stack is growing. Especially for the IPv6 one. It depends on what you wanna build really. What is you target platform by the way? Are you gonna implement RIME or IPv6?

**Comment by LittleFatHero** [comment]
My platform is STM8L151C8T6 whose RAM is  4KB and ROM is 64KB, I only implement RIME and the wireless chip is the popular LoRa chip SX1278. So which version should I choose, contiki 2.7 or contiki 3.x ?
Thanks for answer my question.

**Comment by ekawahyu** [comment]
Porting LoRa over Contiki OS from scratch? Which compiler? SDCC?

**Comment by LittleFatHero** [comment]
> Porting LoRa over Contiki OS from scratch? Which compiler? SDCC?

IAR

**Comment by slcmcu** [comment]
用哪个版本都可以吧

**Comment by ekawahyu** [comment]
Any version can be used, cause you basically have to create the IAR projects by yourself. It is not supported by default in contiki

**Comment by LittleFatHero** [comment]
我是担心3.x版本运行时占用的内存太大了，没办法在STM8L151C8T6上运行 @slcmcu @ekawahyu

**Comment by LittleFatHero** [comment]
中文直接答，牛逼呀 23333 @slcmcu

**Comment by ekawahyu** [comment]
@BigBigWolf-AI If you stick with RIME, that STM8 has plenty. It is not sufficient if you want to use IPv6.

By the way, LoRa has duty cycle limitation, you have very little bandwidth if you have large data to send.

**Comment by LittleFatHero** [comment]
Thank You！ @ekawahyu

**Comment by slcmcu** [comment]
英语不好,没办法只能说中文了,如果看不懂的话,自己找个翻译软件翻译一下吧

**Comment by slcmcu** [comment]
占用内存的问题,最好实际测试一下,毕竟两个版本差别不是很大，就算是移植很不是很难。

**Comment by LittleFatHero** [comment]
谢谢 @slcmcu

**Comment by slcmcu** [comment]
如果想使用IAR的话,就需要移植了,不过在linux编译如果可以的话就不用了

**Comment by LittleFatHero** [comment]
不用IAR, IAR真的是个大坑，不同IAR版本之间创建的代码不兼容。一个博主的移植文章，下载下来他的代码，运行，WTF，70多个错，直接放弃，现在在用Keil，超级好用

**Comment by slcmcu** [comment]
@BigBigWolf-AI  你现在也在做contiki开发吗,用的是什么芯片

**Comment by slcmcu** [comment]
如果GCC支持stm8的话,最好在linux下开发,这样方便升级和维护。

**Comment by LittleFatHero** [comment]
用的STM32, 目前并不知道如何在Linux开发STM32,有比较好的教程或者工具吗？
@slcmcu

**Comment by LittleFatHero** [comment]
以后会用STM8或者STM32L系列，主要是为了低功耗
@slcmcu

**Comment by ekawahyu** [comment]
Any 8-bit MCU like 8051 or STM8 is pain to develop, especially since no GCC support. I would recommend STM32 family, or ARM in general.

**Comment by LittleFatHero** [comment]
Got it, thanks. But the STM8 can save a lot of money for company, so, even pain also have to try! @ekawahyu

**Comment by ekawahyu** [comment]
Agree! Which exact part number of STM8 is it? SDCC compiler is usable for not stable, Cosmic C compiler is free for STM8 without limitation, but I have never tried it.
